src/app/models/test-result.model.ts

```diff
@@ -0,0 +1,97 @@
+import { BodyType } from './body-type';
+import { TestType } from './test.type';
+
+/*
+export interface TestResultModel {
+  reasonForCreation: string;
+  testResultId: string;
+  testerStaffId: string;
+  testStartTimestamp: string;
+  odometerReadingUnits: string;
+  testEndTimestamp: string;
+  testStatus: string;
+  numberOfWheelsDriven: number;
+  testTypes: TestType[];
+  vehicleClass: BodyType;
+  vin: string;
+  vehicleSize: string;
+  testStationName: string;
+  vehicleId: string;
+  vehicleType: string;
+  countryOfRegistration: string;
+  preparerId: string;
+  preparerName: string;
+  odometerReading: number;
+  vehicleConfiguration: string;
+  testStationType: string;
+  reasonForCancellation: string;
+  testerName: string;
+  vrm: string;
+  testStationPNumber: string;
+  numberOfSeats: number;
+  noOfAxles: number;
+  testerEmailAddress: string;
+  euVehicleCategory: string;
+  trailerId: string;
+  testResult: string;
+  systemNumber: string;
+  regnDate: string;
+  firstUseDate: string;
+  deletionFlag: boolean;
+  testExpiryDate: string;
+  testVersion: string;
+  createdByName: string;
+  createdAt: string;
+  lastUpdatedByName: string;
+  lastUpdatedAt: string;
+  testHistory: TestResultModel[];
+  vehicleSubclass: string[];
+}
+*/
+
+export interface TestResultModel {
+  testResultId: string;
+  systemNumber: string;
+  testerStaffId: string;
+  testStartTimestamp: string | Date;
+  odometerReadingUnits: string;
+  testEndTimestamp: string | Date;
+  testStatus: string;
+  testTypes: any[]; //TODO
+  vehicleClass: BodyType;
+  vehicleSubclass?: string[];
+  vin: string;
+  vehicleSize?: string; // Mandatory for PSV only & not applicable to HGV and TRL
```

Comment:
DO we want to bring these models in as is? They aren't in the best of states. I would expect mandatory for PSV/trailer fields etc to be handled by subclasses